avi/resource_avi_cloud.go

```diff
@@ -248,8 +349,27 @@ func resourceAviCloudCreate(d *schema.ResourceData, meta interface{}) error {
 func resourceAviCloudUpdate(d *schema.ResourceData, meta interface{}) error {
 	s := ResourceCloudSchema()
 	var err error
-	err = ApiCreateOrUpdate(d, meta, "cloud", s)
-	if err == nil {
+	cloudType := d.Get("vtype")
+	vcenterConfig, isVcenterConfig := d.GetOk("vcenter_configuration")
+	if cloudType == "CLOUD_VCENTER" && isVcenterConfig {
```

Comment:
we should not have this code duplicated. Let us refactor such that create and update use the same method.